src/api/wallet.go

```diff
@@ -790,16 +788,73 @@ func walletSeedHandler(gateway Gatewayer) http.HandlerFunc {
 			return
 		}
 
-		v := struct {
-			Seed string `json:"seed"`
-		}{
+		v := SeedVerificationReq{
 			Seed: seed,
 		}
 
 		wh.SendJSONOr500(logger, w, v)
 	}
 }
 
+// SeedVerificationReq Request struct for seed
+type SeedVerificationReq struct {
+	Seed string `json:"seed"`
+}
+
+// SeedVerificationResp Response struct for bip39 seed verification
+type SeedVerificationResp struct {
+	IsValid bool `json:"valid"`
+}
+
+// Returns whether the given bip39 Seed is valid or not
+// URI: /api/v2/wallet/seed/verify
+// Method: POST
+// Args:
+//     seed: wallet seed/mnemonic
+func walletVerifySeedHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			resp := NewHTTPErrorResponse(http.StatusMethodNotAllowed, "")
+			writeHTTPResponse(w, resp)
+			return
+		}
+
+		if r.Header.Get("Content-Type") != "application/json" {
+			resp := NewHTTPErrorResponse(http.StatusUnsupportedMediaType, "")
+			writeHTTPResponse(w, resp)
+			return
+		}
+
+		var mn SeedVerificationReq
+		if err := json.NewDecoder(r.Body).Decode(&mn); err != nil {
+			resp := NewHTTPErrorResponse(http.StatusBadRequest, err.Error())
+			writeHTTPResponse(w, resp)
+			return
+		}
+
+		if mn.Seed == "" {
+			resp := NewHTTPErrorResponse(http.StatusBadRequest, "missing seed")
+			writeHTTPResponse(w, resp)
+			return
+		}
+
+		mnemonicValid := bip39.IsMnemonicValid(mn.Seed)
+		if !mnemonicValid {
+			resp := NewHTTPErrorResponse(http.StatusBadRequest, "seed is not a valid bip39 seed")
```

Comment:
Use `StatusUnprocessableEntity`, same as `/api/v2/address/verify`, and update the api/README.md entry similarly